### PR TITLE
Remove state provider

### PR DESCRIPTION
**Pull request by bonndan**
This implements #37 

I don't expect to you to fully understand the logic, it is sufficient to review the code for mistakes.

### CODE REVIEW

**Comment by basseur** [comment]
> While I cannot yet comprehend the logic, I also cannot find any obvious mistakes either.

Btw: I ran the program and it worked. I read all the files and checked the file changes. Anything, in particular, I should have been looking for?

**Comment by bonndan** [comment]
There is an example File containing rancher1 Prometheus Export Data. I have served that file through nginx and it worked as expected. No more to do Herr.

**Comment by bonndan** [comment]
> Looks reasonable, although I do not understand all the details 
> 
> The code contains an alarming number of manual type casts, which you might get rid of by improving the signatures with generic method parameters.
> 
> `Items` contains a lot of static methods. Breaking these up code improve readability. E.g.
> 
> ```
> var filtered = Items.of(landscape).findAll(criteria);
> ```
> 
> By returning `Items`, chaining would be possible and simplify working with subsets:
> 
> ```
>  var filtered = Items.of(landscape).findAll(criteria).findAll(anotherCriteria);
> ```
> 
> Eventually, `Items` might evolve into type-safe collection/query engine for Landscape items.

Yes, a lot of code can be refactored, since some of it was used when JPA persistence was in place.

File: src/main/java/de/bonndan/nivio/model/Items.java [hunk 1]
```diff
@@ -71,28 +71,28 @@ public static LandscapeItem pick(final String identifier, String group, final Co
...
             fqi = FullyQualifiedIdentifier.from(identifier);
         else
             fqi = FullyQualifiedIdentifier.build(null, group, identifier);
-        return findAll(fqi, serviceList);
+        return findAll(fqi, items);
     }
 
-    private static List<LandscapeItem> findAll(FullyQualifiedIdentifier fqi, Collection<? extends LandscapeItem> serviceList) {
-        return serviceList.stream()
+    private static List<LandscapeItem> findAll(FullyQualifiedIdentifier fqi, Collection<? extends LandscapeItem> items) {
```
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/model/Items.java]: I have thought about using Items as non-static wrapper around LandscapeItem collections. This still seems a reasonable idea, especially with the suggested type improvements.

File: src/main/java/de/bonndan/nivio/input/rancher1/PrometheusExporter.java [hunk 2]
```diff
@@ -38,9 +35,9 @@ public PrometheusExporter(String landscape, URL target) {
         this.target = target;
     }
 
-    public Map<FullyQualifiedIdentifier, StatusItem> getStates() {
+    public List<ItemDescription> getDescriptions() {
         PrometheusScraper prometheusScraper = getScraper();
-        final Map<FullyQualifiedIdentifier, StatusItem> tmp = new HashMap<>();
+        final Map<FullyQualifiedIdentifier, ItemDescription> tmp = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/input/rancher1/PrometheusExporter.java]: No, not intentionally. This state was necessary before when the class was responsible for collecting statuses only. The whole class could be merged with the ItemDescriptionFactory for rancher/prometheus.

File: src/main/java/de/bonndan/nivio/input/dto/SourceFormat.java [hunk 3]
```diff
@@ -8,12 +8,20 @@
...
+    NIVIO("nivio", ""),
+    RANCHER1_PROMETHEUS("rancher1-prometheus");
 
     private final List<String> formats;
 
+    private static final List<SourceFormat> KNOWN_FORMATS;
+
+    static {
+        KNOWN_FORMATS = Arrays.asList(NIVIO, DOCKER_COMPOSE2, KUBERNETES, RANCHER1_PROMETHEUS);
+    }
```
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/input/dto/SourceFormat.java]: The current implementation is soon obsolete. #14 will replace it with something happening at runtime.

File: src/main/java/de/bonndan/nivio/model/Items.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/model/Items.java]: I have created #40 from this comment.

**Pull request merged**